### DISCUSSION TITLE
silent uninstall

### DISCUSSION THREAD

**Discussion by KRAER** [valinet/ExplorerPatcher#3971] [Ideas]
I guess a lot of administrator of endpoints ruinning windows 11 are tired of black screens with explorer patcher.

Explorer Patcher has its place for sure, but not on workstations that are used by normal people who do not understand the structure of filesystems nor know the difference between 32 and 64 bit os.

As a system administrator for a few thousand endpoints I really enjoy features that can be installed and uninstalled silently, remotely via rmm tools as SYSTEM account. I can not really wrap my head around why we got this software in our golden images at the first point, but now that we need to uninstall it on the endpoints we need solutions that do not require us in person to log in and hit uninstall and reboot.

If anything fails I will manually remove things and delete autostarts and dll via scripts but to drop some windows that needs interaction in the installer should be the better way to service these problems.

I have checked the FAQ and the last requests regarding silent actions, but could not finde a solution.

Maybe this time it will be integradted into the installer package.

regards
KRAER

**Comment by pyrates999** [comment]
having EP be signed would be a start to this: https://github.com/valinet/ExplorerPatcher/discussions/2353

**Comment by tergalphantom** [comment]
Hi,

I was facing this as well and my last solution was to use AutoIt to catch the "Are you sure to uninstall Explorer Patcher" window and click yes automatically, twice because of the goodbye window. It's awful but it works.

**Comment by fabiankrucker** [comment]
unfortunately had the tool deployed to many workstations. Uninstall works almost silent when using the parameter /uninstall_silent

you can use both:
ep_setup.exe /uninstall_silent
ep_uninstall.exe /uninstall_silent  (ep_uninstall.exe is just a copy of ep_setup.exe which uninstalls the tool)

**Reply by pyrates999** [reply]
This is not mentioned here: https://github.com/valinet/ExplorerPatcher/wiki/Installer-How-To

This should be updated to indicate that.  It is referenced in the file `ep_setup/ep_setup.c`
